pkg/engine/context/resolvers/informerCache.go

```diff
@@ -0,0 +1,92 @@
+package resolvers
+
+import (
+	"context"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+func getSelectorForCache() (labels.Selector, error) {
+	expectedLabels := "cache.kyverno.io/enabled"
+	selector := labels.Everything()
+	requirement, err := labels.NewRequirement(expectedLabels, selection.Exists, nil)
+	if err == nil {
+		selector = selector.Add(*requirement)
+	}
+	return selector, err
+}
+
+func GetResourceInformer(kubeClient kubernetes.Interface, resyncPeriod time.Duration) (kubeinformers.SharedInformerFactory, corev1listers.ConfigMapLister) {
+	selector, err := getSelectorForCache()
+	if err == nil {
+		labelOptions := kubeinformers.WithTweakListOptions(func(opts *metav1.ListOptions) {
+			opts.LabelSelector = selector.String()
+		})
+		kubeResourceInformer := kubeinformers.NewSharedInformerFactoryWithOptions(kubeClient, resyncPeriod, labelOptions)
+		configMapLister := kubeResourceInformer.Core().V1().ConfigMaps().Lister()
+		return kubeResourceInformer, configMapLister
+	}
+	return nil, nil
+}
+
+type NamespacedResourceResolver[T any] interface {
+	Get(context.Context, string, string) (T, error)
+}
+
+type ConfigmapResolver = NamespacedResourceResolver[*corev1.ConfigMap]
+
+type informerBasedResolver struct {
+	lister corev1listers.ConfigMapLister
+}
+
+func NewInformerBasedResolver(lister corev1listers.ConfigMapLister) ConfigmapResolver {
```

Comment:
What if list is nil ?

Reply:
we are calling `Get` to informer or kubeclient, so if we don't find CM, it will throw error

Reply:
handled `nil` in `ResolverChain` this function